CommonMC/src/EventWindowMarkerProducer_module.cc

```diff
@@ -38,6 +38,8 @@ namespace mu2e {
         fhicl::Atom<unsigned> spillType { Name( "SpillType"), Comment("Whether simulating on or off spill")};
         fhicl::Atom<bool> recoFromMCTruth{ Name("RecoFromMCTruth"), Comment("Create reco PBT from MC truth"),false};
         fhicl::Atom<float> recoFromMCTruthErr{ Name("RecoFromMCTruthErr"), Comment("If creating reco PBT from MC truth, smear by this amount in ns"),0};
+        fhicl::Atom<bool> fixInitialPhase{ Name("FixInitialPhase"), Comment("If false will randomize phase every run"),false};
+        fhicl::Atom<float> initialPhaseShift{ Name("InitialPhaseShift"), Comment("Initial phase shift between beam and detector clocks in fraction of one clock tick."),0};
```

Comment:
Andrei has pointed out a better design pattern for this kind of config, namely OptionalAtom.   In this case initialPhaseShift would become an OptionalAtom, and the bool Atom would go away.  The test on the bool would be replaced by a test whether the OptionalAtom is set or not.  This combines the config and prevents ambiguous situations (ie what to do if someone sets fixInitialPhase to false but doesn't provide a value, etc).  An example is

Reply:
That's a lot more convenient, good to know that exists!